### PR TITLE
Add conditional render for PreSubmitSection

### PR DESCRIPTION
**Pull request by LinnJS**
## Description
- Add conditional render for PreSubmitSection, if customComponent is passed that component is rendered instead of the default `PreSubmitSection`
-  Updated documentation to reflect changes in link: [Documentation PR](https://github.com/department-of-veterans-affairs/veteran-facing-services-tools/pull/217)
- Documented conditional render function in a code comment

## Testing done
- Manually tested with another branch to confirm working as expected
- Added unit test to `SubmitController` to cover both sides of the ternary. 

## Screenshots
Screen Shot 2020-03-09 at 11 41 31 AM

## Acceptance criteria
- [x] If customComponent is passed render that component instead of default `PreSubmitSection`

## Definition of done
- [x] If customComponent is passed render that component instead of default `PreSubmitSection`
- [x] Unit test coverage

### CODE REVIEW

File: src/platform/forms-system/src/js/review/SubmitController.jsx [hunk 1]
```diff
@@ -30,6 +30,7 @@ class SubmitController extends React.Component {
     field: 'AGREED',
     label: 'I agree to the terms and conditions.',
     error: 'You must accept the agreement before submitting.',
+    CustomComponent: false,
```
Comment (cvalarida) [inline src/platform/forms-system/src/js/review/SubmitController.jsx]:
```suggestion
```
We can leave this out. `undefined` is a bit more semantic than `false`.

**Review by cvalarida — COMMENTED** [review]
Looks great! Only thing left I'd say that needs to be done here is adding a couple tests.
1. `it('should render the default PreSubmitSection', () => {});`
2. `it('should render a custom component', () => {});`

**Review by cvalarida — APPROVED** [review]
Looks great. Thanks for adding those tests! :raised_hands:

**Review by Mottie — APPROVED** [review]
I was a little thrown off by the changes since I just recently updated this file, but now I understand!

It looks good to me!

File: src/platform/forms-system/src/js/components/PreSubmitSection.jsx [hunk 2]
```diff
@@ -30,7 +30,7 @@ export function PreSubmitSection({
 }
 
 PreSubmitSection.propTypes = {
-  onChange: PropTypes.func.isRequired,
+  sectionCompleted: PropTypes.func.isRequired,
   preSubmitInfo: PropTypes.object.isRequired,
   showError: PropTypes.bool,
 };
```
Comment (mr0sari0) [inline src/platform/forms-system/src/js/components/PreSubmitSection.jsx]: Awesome job with adding propTypes and I like the semantic change from `onChange` to `sectionCompleted`; that will be very helpful for future usage of this component.

**Review by maharielrosario — APPROVED** [review]
Overall this is a great feature for anyone wanting to use custom components for pre-submits. Good inline documentation as well.

File: src/platform/forms-system/src/js/components/PreSubmitSection.jsx (hunk 2 above)
Comment (LinnJS) [reply src/platform/forms-system/src/js/components/PreSubmitSection.jsx]: Thank you was suggested by @cvalarida, cant take all the credit 😉

**Pull request merged**